### DISCUSSION TITLE
yaml file after building app

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#24990] [Q&A General] [answered]
Recently updated my local copy of the framework and now my app produces an <appname>.yaml "resource file" when compiling. What's that used for? I'm inclined to toss it in my .gitignore.

**Comment by milljm** [comment, marked as answer]
It's a dynamically generated file containing information about the build. It has potential for other things we might want to track down the road. But for now, the TestHarness uses it to understand if your build is an in-tree build, or out-of-tree build (like an install or not).

It's safe to add this file to your .gitignore.

Edit:
Test Spec file making use of this feature:

```pre
[Tests]
  [./in_tree_type]
    type = RunCommand
    command = "true"
    # Test will only run if application was built in-tree
    installation_type = IN_TREE
  [../]
  [./installed_type]
    type = RunCommand
    command = "true"
    # Test will only run if application is installed
    installation_type = INSTALLED
  [../]
  [./all_type]
    type = RunCommand
    command = "true"
    # Test will always run (default)
    installation_type = ALL
  [../]
[]
```

**Reply by aarograh** [reply]
Hello,

Piggybacking off this question, I have a couple of other apps that are coupled to my app.  I've set up the `tests` files like so:
```
[Tests]
  [griffin_standalone]
    type = Exodiff
    input = griffin_standalone.i
    exodiff = griffin_standalone_out.e
    group = 'griffin_coupled'
    required_applications = GriffinApp
  []
[]
```
for each of the tests.

However, when I run the tests, I get this:
```
test:coupled/griffin/full_coupling.griffin_standalone ..... [App GriffinApp not registered in executable] SKIP
```

I see this in my app's .yml file:
```
registered_apps:
- WASPAPP
- MOLEAPP
- MOLETESTAPP
```

I've confirmed that the `tests` file is reading the `.yaml` file to find `GRIFFINAPP` and `SAMAPP` and then skipping those tests when it does not see them (I manually added them to the `.yaml` file to confirm this).  Obviously there is some sort of registration step I'm missing, but I don't know what it is.  Could you help me understand how to ensure that griffin and SAM apps are registered as well?

**Reply by GiudGiud** [reply]
you can register apps in the XXXApp.C file 
```
void
MyApp::registerApps()
{
  registerApp(MyApp);
#ifdef GRIFFIN_ENABLED
  GriffinApp::registerApps();
#endif
```

**Reply by aarograh** [reply]
Thank you for the response.  I actually already had that in there.  After grepping around the guts of MOOSE I figured out that the issue was that `ADDITIONAL_CPPFLAGS += -DGRIFFIN_ENABLED` in the `Makefile` should have been `libmesh_CXXFLAGS += -DGRIFFIN_ENABLED`.  Once I made that change it worked correctly

**Reply by GiudGiud** [reply]
> ADDITIONAL_CPPFLAGS += -DGRIFFIN_ENABLED in the Makefile should have been libmesh_CXXFLAGS += -DGRIFFIN_ENABLED

that does not seem right but if it works let s go
